### PR TITLE
BITMAG-1175: jQuery update

### PR DESCRIPTION
**Pull request by Bohlski**
Besides updating jQuery I decided to fiddle with the webapp directory structure in bitrepository-webclient along with doing some general housecleaning in the module.

### CODE REVIEW

File: bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.resize.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ plots, you can just fix the size of their placeholders.
...
+(function($,h,c){var a=$([]),e=$.resize=$.extend($.resize,{}),i,k="setTimeout",j="resize",d=j+"-special-event",b="delay",f="throttleWindow";e[b]=250;e[f]=true;$.event.special[j]={setup:function(){if(!e[f]&&this[k]){return false}var l=$(this);a=a.add(l);$.data(this,d,{w:l.width(),h:l.height()});if(a.length===1){g()}},teardown:function(){if(!e[f]&&this[k]){return false}var l=$(this);a=a.not(l);l.removeData(d);if(!a.length){clearTimeout(i)}},add:function(l){if(!e[f]&&this[k]){return false}var n;function m(s,o,p){var q=$(this),r=$.data(this,d);r.w=o!==c?o:q.width();r.h=p!==c?p:q.height();n.apply(this,arguments)}if(typeof l==="function"){n=l;return m}else{n=l.handler;l.handler=m}}};function g(){i=h[k](function(){a.each(function(){var n=$(this),m=n.width(),l=n.height(),o=$.data(this,d);if(m!==o.w||l!==o.h){n.trigger(j,[o.w=m,o.h=l])}});g()},e[b])}})(jQuery,this);
```
Comment (m-atlantis) [inline bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.resize.js:22]: Why is this a one-liner? 👀

File: bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.selection.min.js [hunk 2]
```diff
@@ -1 +1 @@
...
...u.axis.p2c(u.to)}if(w.selection.mode=="x"){p.first.y=0;p.second.y=k.height()}else{u=c(t,"y");p.first.y=u.axis.p2c(u.from);p.second.y=u.axis.p2c(u.to)}p.show=true;k.triggerRedrawOverlay();if(!s&&f()){i()}}function f(){var s=5;return Math.abs(p.second.x-p.first.x)>=s&&Math.abs(p.second.y-p.first.y)>=s}k.clearSelection=q;k.setSelection=o;k.getSelection=g;k.hooks.bindEvents.push(function(t,s){var u=t.getOptions();if(u.selection.mode!=null){s.on("mousemove",e);s.on("mousedown",n)}});k.hooks.drawOverlay.push(function(v,D){if(p.show&&f()){var t=v.getPlotOffset();var s=v.getOptions();D.save();D.translate(t.left,t.top);var z=a.color.parse(s.selection.color);D.strokeStyle=z.scale("a",0.8).toString();D.lineWidth=1;D.lineJoin="round";D.fillStyle=z.scale("a",0.4).toString();var B=Math.min(p.first.x,p.second.x),A=Math.min(p.first.y,p.second.y),C=Math.abs(p.second.x-p.first.x),u=Math.abs(p.second.y-p.first.y);D.fillRect(B,A,C,u);D.strokeRect(B,A,C,u);D.restore()}});k.hooks.shutdown.push(function(t,s){s.off("mousemove",e);s.off("mousedown",n);if(r){a(document).off("mouseup",r)}})}a.plot.plugins.push({init:b,options:{selection:{mode:null,color:"#e8cfac"}},name:"selection",version:"1.1"})})(jQuery);
```
Comment (m-atlantis) [inline bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.selection.min.js:1]: This one as well? Is it just a GitHub thing?

**Review by m-atlantis — COMMENTED** [review]
Love the changes! Just had that question about why that function is a one-liner, apart from that I it's looking beautiful!

File: bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.resize.js (hunk 1 above)
Comment (Bohlski) [reply bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.resize.js:22]: No clue why but some of the flot library files simply contain both minified and regular code. I think it's just some weird legacy thing.

File: bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.selection.min.js (hunk 2 above)
Comment (Bohlski) [reply bitrepository-webclient/src/main/webapp/static/js/flot/jquery.flot.selection.min.js:1]: In this case the file just contains only minified script. So no whitespace = smaller file to deliver to browser.

I just made a small change in it to make it comply with the jQuery update.

**Pull request merged**